### PR TITLE
feat: wherever widget

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

* adds new Wherever notification widget - allowing users to subscribe to contracts they want notifications for, via multiple channels, e.g. email, telegram
* please see original PR - taking over from https://github.com/shapeshift/web/pull/2791
* new dependency, please note this a very new SDK - https://github.com/whereverlabs/react-notification-widget

remaining

* [x] use prod key everywhere? awaiting response from @nivo33 

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

n/a

## Risk

fairly small and flaggable - new feature and SDK, isolated to EVM chains

## Testing

* this will only work with metamask/walletconnect, i.e https://eips.ethereum.org/EIPS/eip-1193 enabled wallets. discussing with engineering how we can put our own 1193 provider into the stack to make hdwallet more of an "online"/metamask-esque type wallet.

### Engineering

* **analytics** - check localstorage that we don't see any k/v pairs including `rudder` - this is a third party analytics stack that wherever uses upstream. we've disabled analytics on their sdk and i've verified this works and doesn't reach out to rudder.

### Operations

During phase 1 (now):
1. We'll need a dummy welcome message set up
2. Sign up with your wallet, and opt-in to the available options: Discord, telegram, email
3. once you opt-in confirm that you get the welcome message on those channels
4. Report any wonky UX
5. Create a one-off message to send to your wallet and other test wallets that have FOX in them. For example, you should be able to search and find the wallets with FOX and then send out a message to those wallets like "Stake your FOX to earn up to 4% APY" or similar

During phase 2 (in two weeks)
1. We'll be able to test on chain activity
2. Buy some ETH through OnRamper or wherever (and send yourself an ethereum based token to the wallet you're testing)
3. Confirm that you're notified once the asset hits your wallet

Reach out to @DiggyDiggy2 or @willyogo with any questions around testing this

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -121,6 +121,7 @@
     "@visx/xychart": "^2.12.2",
     "@wagmi/core": "^0.5.5",
     "@walletconnect/web3-provider": "^1.7.8",
+    "@wherever/react-notification-feed": "^0.0.11",
```
Comment (0xdef1cafe) [inline package.json:124]: @nivo33 can we please see the source code for this before we integrate it? i can see it on npm but no link to github https://www.npmjs.com/package/@wherever/react-notification-feed?activeTab=versions

Comment (0xdef1cafe) [reply package.json:124]: i can see this contains analytics with hardcoded keys

**Comment by 0xdef1cafe** [comment]
@reallybeard two things

1. do we want displaying in the header on mobile?
2. can we get some vertical padding on the modal when it's open so it sits down from the header a lil?

File: src/components/Layout/Header/NavBar/Notifications.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  const disableAnalytics = window.location.hostname.includes('private.shapeshift.com')
+  const partnerKey = getConfig().REACT_APP_WHEREVER_PARTNER_KEY
+
+  return (
+    <Box>
+      <NotificationFeedProvider
+        partnerKey={partnerKey}
+        theme={themeObj}
+        disableAnalytics={disableAnalytics}
```
Comment (0xdef1cafe) [inline src/components/Layout/Header/NavBar/Notifications.tsx:58]: as discussed via DM - the library is not respecting this flag and still trying instantiating analytics when this is set to true

https://github.com/whereverlabs/react-notification-widget/blob/staging/src/services/analytics.ts#L50

File: src/components/Layout/Header/NavBar/Notifications.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+} from '@wherever/react-notification-feed'
+import { getConfig } from 'config'
+import { useMemo } from 'react'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { getLocalWalletType } from 'context/WalletProvider/local-wallet'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { breakpoints, theme } from 'theme/theme'
+
+const WHEREVER_ENABLED_WALLETS: KeyManager[] = [
+  KeyManager.TallyHo,
```
Comment (0xdef1cafe) [inline src/components/Layout/Header/NavBar/Notifications.tsx:16]: currently only enabled for EIP-1193 wallets

https://eips.ethereum.org/EIPS/eip-1193

at some point, probably soon, we're going to have to add this concept to hdwallet for native and keepkey wallets so we're more metamask-like

Comment (gomesalexandre) [reply src/components/Layout/Header/NavBar/Notifications.tsx:16]: Got this one for ya if we feel like writing a type guard https://github.com/shapeshift/web/blob/4e8c9ca946e3d9cef138a714efb029f868aa8b84/src/context/WalletProvider/WalletProvider.tsx#L90-L95

File: src/components/Layout/Header/NavBar/Notifications.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  KeyManager.MetaMask,
+  KeyManager.WalletConnect,
+]
+
+export const Notifications = () => {
+  const isWhereverEnabled = useFeatureFlag('Wherever')
+  const { colorMode } = useColorMode()
+
+  const currentWallet = getLocalWalletType()
+  const mobileBreakpoint = Number(breakpoints.md.replace('px', ''))
```
Comment (gomesalexandre) [inline src/components/Layout/Header/NavBar/Notifications.tsx:26]: 🧠

**Review by gomesalexandre — APPROVED** [review]
Retested on a different browser profile with TallyHo, confirmed doing what it says on the box and vertical alignment is fixed

[](image.png)

**Pull request merged**